### PR TITLE
Update nomad-deploy-app script with workable template.

### PR DESCRIPTION
**Pull request by cconstantine**
Potential kairos data file:
``` yaml
app_name: Kairos
deploy_env: staging
cluster: dc1
groups:
  - name: "web"
    count: 2
    tasks:
      - name: "puma"
        docker:
          image: "foo"
          command: "puma"
          args:
            - "-e"
            - "staging"
          port_map:
            puma: 3000
        resources:
          cpu: 100
          memory: 500
          network:
            mbits: 100
            ports:
              - puma
        services:
          - name: kairos
            port: puma
            check:
              type: http
              port: puma
              path: "/health_check"
              interval: 10s
              timeout: 60s
        env:
          DEPLOY_ENV: "STAGING"
          RAILS_ENV: "STAGING"
          APOHYPATON_URL: "{{ 'CONSUL_URL' | env }}"
          KAIROS_CONSUL_TOKEN: "{{ 'CONSUL_TOKEN' | env | secret('CONSUL_TOKEN') }}"
          RAILS_SERVE_STATIC_FILES: "true"
          RAILS_LOG_TO_STDOUT: "true"


  - name: "jobs"
    count: 2
    tasks:
      - name: "sidekiq"
        docker:
          image: "foo"
          command: "sidekiq"
          args:
            - "-e"
            - "staging"
        resources:
          cpu: 100
          memory: 500
          network:
            mbits: 100
        services:
          - name: sidekiq
        env:
          DEPLOY_ENV: "STAGING"
          RAILS_ENV: "STAGING"
          APOHYPATON_URL: "{{ 'CONSUL_URL' | env }}"
          KAIROS_CONSUL_TOKEN: "{{ 'CONSUL_TOKEN' | env | secret('CONSUL_TOKEN') }}"
          RAILS_SERVE_STATIC_FILES: "true"
          RAILS_LOG_TO_STDOUT: "true"
```

### CODE REVIEW

**Comment by cconstantine** [comment]
Another option is to drop this project entirely in favor of https://github.com/jrasell/levant

File: nomad_deploy/templates/service_app.hcl.j2 [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        ]
+        {%- endif %}
+
+        {%- if task.docker.port_map is defined %}
+        port_map {
+        {%- for name, port in task.docker.port_map.items() %}
+          {{ name }} = {{ port }}
+        {%- endfor %}
+        }
+        {%- endif %}
```
Comment (jimmyatwork) [inline nomad_deploy/templates/service_app.hcl.j2]: Ethel also needs a way to bind-mount a directory in the EFS share:
https://github.com/omadahealth/ethel/blob/master/nomad.hcl.j2#L37

Comment (cconstantine) [reply nomad_deploy/templates/service_app.hcl.j2]: Yup, that needs to be added in.

File: nomad_deploy/templates/service_app.hcl.j2 [hunk 2]
```diff
@@ -0,0 +1,103 @@
+job "{{app_name}}/{{deploy_env}}" {
+  datacenters = ["{{ cluster }}"]
```
[the remark this answers is not in the document]
Comment (cconstantine) [reply nomad_deploy/templates/service_app.hcl.j2]: Oh, ,maybe / probably.

**Comment by cconstantine** [comment]
The more I think about it, the more I like levant.  The only feature it's really missing is the ability to safely print out the rendered job.

How important do you think it is to see the fully-rendered template in something like the jenkins deploy job (where non-redacted secrets in the template might be a problem)?

**Comment by cconstantine** [comment]
Oh yeah, and we'd also miss having a standard job template; each app would have its own template.

File: nomad_deploy/templates/service_app.hcl.j2 (hunk 2 above)
[the remark this answers is not in the document]
Comment (cconstantine) [reply nomad_deploy/templates/service_app.hcl.j2]: Added

File: nomad_deploy/templates/service_app.hcl.j2 (hunk 1 above)
Comment (cconstantine) [reply nomad_deploy/templates/service_app.hcl.j2]: added

File: nomad_deploy/__init__.py [hunk 3]
```diff
@@ -1,55 +1,69 @@
...
+    try:
+      temp_file, tempfile_name = tempfile.mkstemp()
+
+      os.write(temp_file, self.template.render_hcl())
+      os.close(temp_file)
+
+      if print_template:
+        print("======== Job ========")
+        template_stream = StringIO.StringIO(self.template.render_hcl_safe())
+        line_num = 1
```
Comment (jimmyatwork) [inline nomad_deploy/__init__.py]: Looks like `line_num` is unused.

File: nomad_deploy/__init__.py [hunk 4]
```diff
@@ -1,55 +1,69 @@
...
+def app():
+  parser = argparse.ArgumentParser(description="Run a nomad application.")
   parser.add_argument('--show-job', '-s', default=False , action='store_true')
-  parser.add_argument('template', metavar='template_file', type=file, nargs=1,
-    help="Nomad job definition template")
+  parser.add_argument('--dry-run', '-n', default=False , action='store_true')
   parser.add_argument('data', metavar='data_file', type=file, nargs="+",
-    help="Date for template rendering")
-
+    help="Application definintion file (yaml)")
```
Comment (jimmyatwork) [inline nomad_deploy/__init__.py]: `definintion` typo

File: nomad_deploy/__init__.py [hunk 5]
```diff
@@ -1,55 +1,69 @@
...
-
-  sys.exit(job_exec)
+def job():
+  parser = argparse.ArgumentParser(description="Run a nomad job file.")
+  parser.add_argument('--show-job', '-s', default=False , action='store_true')
+  parser.add_argument('--dry-run', '-n', default=False , action='store_true')
+  parser.add_argument('template', metavar='template_file', type=file, nargs=1,
+    help="Nomad job definition template")
+  parser.add_argument('data', metavar='data_file', type=file, nargs="+",
+    help="Date for template rendering")
```
Comment (jimmyatwork) [inline nomad_deploy/__init__.py]: `Date` should probably be `Data`?

**Comment by cconstantine** [comment]
Hmm, yeah.  nomad-deploy *could* call `levant` instead of `nomad`.  That's not a bad idea (I think?)

**Comment by cconstantine** [comment]
I'm going to abandon this line of work and close this PR.

**Pull request closed without merging**